cartography/intel/aws/ec2/instances.py

```diff
@@ -42,17 +42,46 @@
 )
 
 
+@timeit
+def get_ec2_images(boto3_session: boto3.session.Session, image_ids: List[str], region: str) -> Dict[str, Dict]:
+    client = boto3_session.client("ec2", region_name=region, config=get_botocore_config())
+
+    try:
+        response = client.describe_images(ImageIds=image_ids)
+        images = response.get("Images", [])
+
+        # returning a dictionary where the image id is the key and details are values
+        return {image['ImageId']: image for image in images}
+    except ClientError as e:
+        logger.error(f"Error feetching the image details {e}")
+        return {}
+
+
 @timeit
 @aws_handle_regions
 def get_ec2_instances(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
     client = boto3_session.client("ec2", region_name=region, config=get_botocore_config())
     reservations = []
+    image_ids = []
     try:
         paginator = client.get_paginator("describe_instances")
         for page in paginator.paginate():
             reservations.extend(page["Reservations"])
+            for reservation in page["Reservations"]:
+                for instance in reservation["Instances"]:
+                    image_id = instance.get("ImageId")
+                    if image_id:
+                        image_ids.append(image_id)
+
+        image_details = get_ec2_images(boto3_session, list(set(image_ids)), region)
```

Comment:
this need to be optimized. We shouldn't call AWS API for each image for each instance. Imagine a customer with 20K servers, we will be making 20K API calls. Let's batch them into as less calls as possible.

Reply:
Got it! I'll work on it
will push the code for Azure and GCP by tomorrow morning